lib/widgets/witnet/transactions/value_transfer/create_dialog_box/vtt_builder/02_select_miner_fee.dart

```diff
@@ -0,0 +1,356 @@
+      _minerFeeOptions[EstimatedFeeOptions.Opulent] = opulent;
+    });
+  }
+
+  num _minerFeeToNumber() {
+    try {
+      return num.parse(_minerFee != '' ? _minerFee : '0');
+    } catch (e) {
+      return 0;
+    }
+  }
+
+  bool _isAbsoluteFee() {
+    return _feeType == FeeType.Absolute;
+  }
+
+  bool _notEnoughFunds({int? customFee = null}) {
+    final balance = balanceInfo.availableNanoWit;
+    final amount = BlocProvider.of<VTTCreateBloc>(context)
+        .state
+        .vtTransaction
+        .body
+        .outputs
+        .first
+        .value
+        .toInt();
+    if (_feeType == FeeType.Absolute) {
+      int minerFee = customFee ??
+          int.parse(_minerFeeToNumber().standardizeWitUnits(
+              inputUnit: WitUnit.Wit, outputUnit: WitUnit.nanoWit));
+      int totalToSpend = minerFee + amount;
+      return balance < totalToSpend;
+    } else {
+      int? _weightedFee = balanceInfo.weightedVttFee(amount);
+      return _weightedFee != null ? balance < _weightedFee + amount : true;
+    }
+  }
+
+  void _setSavedFeeData() {
+    if (widget.savedFeeType != null) _setFeeType(widget.savedFeeType?.name);
+
+    if (_isAbsoluteFee() && widget.savedFeeAmount != null) {
+      _minerFeeController.text = widget.savedFeeAmount!;
+      _minerFee = widget.savedFeeAmount!;
+      _minerFeeController.text = widget.savedFeeAmount!;
+    }
+    ;
+  }
+
+  void _setFeeType(type) {
+    setState(() {
+      _feeType = type == "Absolute" ? FeeType.Absolute : FeeType.Weighted;
+    });
+  }
+
+  void _updateTxFee() {
+    if (_isAbsoluteFee()) {
+      _setAbsoluteFee();
+    } else {
+      _setWeightedFee();
+    }
+    validateForm();
+  }
+
+  void _setAbsoluteFee() {
+    BlocProvider.of<VTTCreateBloc>(context).add(UpdateFeeEvent(
+        feeType: FeeType.Absolute,
+        feeNanoWit: int.parse(_minerFeeToNumber().standardizeWitUnits(
+            inputUnit: WitUnit.Wit, outputUnit: WitUnit.nanoWit))));
+  }
+
+  void _setWeightedFee() {
+    BlocProvider.of<VTTCreateBloc>(context)
+        .add(UpdateFeeEvent(feeType: FeeType.Weighted));
+  }
+
+  String? _validateFee(String? input) {
+    String? errorText;
+    try {
+      num.parse(_minerFee != '' ? _minerFee : '0');
+    } catch (e) {
+      errorText = 'Invalid Amount';
+    }
+    if (_notEnoughFunds()) {
+      errorText = 'Not enough Funds';
+    }
+    if (_isAbsoluteFee()) {
+      errorText = errorText ?? validateWitValue(input);
+    }
+    return errorText;
+  }
+
+  bool validateForm() {
+    setState(() {
+      _errorFeeText = _validateFee(_minerFee);
+    });
+    return _errorFeeText == null;
+  }
+
+  void nextAction() {
+    if (validateForm() && _formKey.currentState!.validate()) {
+      _updateTxFee();
+    }
+  }
+
+  NavAction next() {
+    return NavAction(
+      label: 'Continue',
+      action: nextAction,
+    );
+  }
+
+  Widget _buildFeeOptionButton(EstimatedFeeOptions label, String value) {
```

Comment:
When there is an error, all borders should be red. Right now, one of them is black.